src/inspect_ai/model/_providers/bedrock.py

```diff
@@ -42,97 +67,56 @@ def __init__(
-            from .anthropic import AnthropicAPI
+        # import aioboto3 on demand
+        try:
+            import aioboto3
 
-            self.model_api = AnthropicAPI(
-                model_name=model_name,
-                base_url=base_url,
-                config=config,
-                bedrock=True,
-                **model_args,
-            )
-        elif is_mistral(model_name):
-            self.handler: BedrockChatHandler = MistralChatHandler(
-                model_name, base_url, config
-            )
-        elif is_llama2(model_name):
-            self.handler = Llama2ChatHandler(model_name, base_url, config)
-        elif is_llama3(model_name):
-            self.handler = Llama3ChatHandler(model_name, base_url, config)
-        else:
-            raise ValueError(f"Unsupported Bedrock model: {model_name}")
+            verify_required_version("Bedrock API", "aioboto3", "13.0.0")
 
-    async def generate(
-        self,
-        input: list[ChatMessage],
-        tools: list[ToolInfo],
-        tool_choice: ToolChoice,
-        config: GenerateConfig,
-    ) -> ModelOutput | tuple[ModelOutput, ModelCall]:
-        if self.model_api:
-            return await self.model_api.generate(input, tools, tool_choice, config)
-        else:
-            return await self.handler.generate(input, tools, tool_choice, config)
+            # Create a shared session to be used when generating
+            self.session = aioboto3.Session()
 
-    @override
-    def max_tokens(self) -> int | None:
-        if self.model_api:
-            return self.model_api.max_tokens()
-        else:
-            return self.handler.max_tokens()
+        except ImportError:
+            raise pip_dependency_error("Bedrock API", ["aioboto3"])
 
     @override
     def connection_key(self) -> str:
         return self.model_name
 
     @override
-    def is_rate_limit(self, ex: BaseException) -> bool:
-        if self.model_api:
-            return self.model_api.is_rate_limit(ex)
-        else:
-            return self.handler.is_rate_limit(ex)
+    def max_tokens(self) -> int | None:
+        if "llama3" in self.model_name or "claude3" in self.model_name:
+            return 4096
 
-    @override
-    def collapse_user_messages(self) -> bool:
-        if self.model_api:
-            return self.model_api.collapse_user_messages()
-        else:
-            return super().collapse_user_messages()
+        elif "mistral-large" in self.model_name:
+            return 8192
 
+        # Other models will just the default
+        else:
+            return DEFAULT_MAX_TOKENS
 
-# https://docs.aws.amazon.com/bedrock/latest/userguide/inference-invoke.html
-class BedrockChatHandler(abc.ABC):
-    def __init__(
-        self, model_name: str, base_url: str | None, config: GenerateConfig
-    ) -> None:
-        # import aioboto3 on demand
-        try:
-            import aioboto3
+    @override
+    def is_rate_limit(self, ex: BaseException) -> bool:
```

Comment:
Is there no way that a 429 ever gets through?

Reply:
ThrottlingException wraps HTTP 429s afaict so we should be good.